nrfutil/device/batch.ts

```diff
@@ -133,21 +154,34 @@ export class Batch {
         count: number,
         callback: (completedTasks: TaskEnd<unknown>[]) => void
     ) {
+        if (!callback || typeof count === 'undefined') {
+            // handle if not passed: log use case or anything else
+            // same check can be applied to the methods above if it makes sense
+
+            return this;
+        }
+
         this.collectOperations.push({
             callback,
+            // I'm not sure if we should set operationId, as it should be global Id for the whole batch operation
+            // per my understanding
+            // also note that if no operations on the list value will be "-1" which might be confusing at least
             operationId: this.operations.length - 1,
             count,
         });
 
         return this;
     }
 
-    public async run(
+    // here would be nice to have an option to pass timeout (like 2/5/10min etc) and allow user to abort operation if stuck
+    // otherwise we might be stuck in infinite await, right?
```

Comment:
_If_ we need a timeout, it is already possible to implement it in the client of the function. They can abort the task at any time. If we need this very often, then we can add it later here.

Reply:
That's an idea I've just shared here. Timeouts are also implemented inside device-lib, so we might not need that at all. I'm just not sure what logic we're losing by dropping device-lib-js.